lib/Transforms/SetSPIRVAbiAttribute.cpp

```diff
@@ -53,31 +53,34 @@ class SetSPIRVAbiAttributePass
   }
 
   void runOnOperation() override {
-    auto gpuModule = getOperation();
     auto *context = &getContext();
     auto attrName =
         mlir::StringAttr::get(context, mlir::spirv::getEntryPointABIAttrName());
-    if (m_clientAPI == "opencl") {
-      auto abi = mlir::spirv::getEntryPointABIAttr(context);
-      for (const auto &gpuFunc : gpuModule.getOps<mlir::gpu::GPUFuncOp>()) {
-        if (!mlir::gpu::GPUDialect::isKernel(gpuFunc) ||
-            gpuFunc->getAttr(attrName))
-          continue;
+    auto op = getOperation();
+    mlir::StringRef clientAPI = m_clientAPI;
+    op->walk([&](mlir::gpu::GPUModuleOp gpuModule) {
```

Comment:
Hi SangIk (@silee2 ),
Thanks a lot for taking care of this. I have question though,
Why would we need to do the walk? Did MLIR change the pass structure somehow?
What I am wondering is, why the old version does not work?